src/client/inputhandler.cpp

```diff
@@ -112,56 +114,78 @@ bool MyEventReceiver::OnEvent(const SEvent &event)
+				// stream reach the application as it will cause
+				// certain actions to repeat constantly.
+				if (!IsKeyDown(keyCode))
+					keyWasDown.set(keyCode);
 				keyIsDown.set(keyCode);
-				keyWasDown.set(keyCode);
 			} else {
 				keyIsDown.unset(keyCode);
 			}
 			return true;
 		}
-	}
 
 #ifdef HAVE_TOUCHSCREENGUI
-	// case of touchscreengui we have to handle different events
-	if (m_touchscreengui && event.EventType == irr::EET_TOUCH_INPUT_EVENT) {
+	} else if (m_touchscreengui && event.EventType == irr::EET_TOUCH_INPUT_EVENT) {
+		// In case of touchscreengui, we have to handle different events
 		m_touchscreengui->translateEvent(event);
 		return true;
-	}
 #endif
 
-	if (event.EventType == irr::EET_JOYSTICK_INPUT_EVENT) {
+	} else if (event.EventType == irr::EET_JOYSTICK_INPUT_EVENT) {
 		/* TODO add a check like:
 		if (event.JoystickEvent != joystick_we_listen_for)
 			return false;
 		*/
 		return joystick->handleEvent(event.JoystickEvent);
-	}
-	// handle mouse events
-	if (event.EventType == irr::EET_MOUSE_INPUT_EVENT) {
-		if (isMenuActive()) {
-			left_active = false;
-			middle_active = false;
-			right_active = false;
-		} else {
-			left_active = event.MouseInput.isLeftPressed();
-			middle_active = event.MouseInput.isMiddlePressed();
-			right_active = event.MouseInput.isRightPressed();
-
-			if (event.MouseInput.Event == EMIE_LMOUSE_PRESSED_DOWN) {
-				leftclicked = true;
-			}
-			if (event.MouseInput.Event == EMIE_RMOUSE_PRESSED_DOWN) {
-				rightclicked = true;
-			}
-			if (event.MouseInput.Event == EMIE_LMOUSE_LEFT_UP) {
-				leftreleased = true;
-			}
-			if (event.MouseInput.Event == EMIE_RMOUSE_LEFT_UP) {
-				rightreleased = true;
-			}
-			if (event.MouseInput.Event == EMIE_MOUSE_WHEEL) {
-				mouse_wheel += event.MouseInput.Wheel;
-			}
+	} else if (event.EventType == irr::EET_MOUSE_INPUT_EVENT) {
+		// Handle mouse events
+		KeyPress key;
+		switch (event.MouseInput.Event) {
+		case EMIE_LMOUSE_PRESSED_DOWN:
+			key = "KEY_LBUTTON";
+			keyIsDown.set(key);
+			keyWasDown.set(key);
+			break;
+		case EMIE_MMOUSE_PRESSED_DOWN:
+			key = "KEY_MBUTTON";
+			keyIsDown.set(key);
+			keyWasDown.set(key);
+			break;
+		case EMIE_RMOUSE_PRESSED_DOWN:
+			key = "KEY_RBUTTON";
+			keyIsDown.set(key);
+			keyWasDown.set(key);
+			break;
+		case EMIE_LMOUSE_LEFT_UP:
+			key = "KEY_LBUTTON";
+			keyIsDown.unset(key);
+			break;
+		case EMIE_MMOUSE_LEFT_UP:
+			key = "KEY_MBUTTON";
+			keyIsDown.unset(key);
+			break;
+		case EMIE_RMOUSE_LEFT_UP:
+			key = "KEY_RBUTTON";
+			keyIsDown.unset(key);
+			break;
+		case EMIE_MOUSE_WHEEL:
+			mouse_wheel += event.MouseInput.Wheel;
+			break;
+
+		// Add cases for all other enum values to silence compiler warning
+		case EMIE_MOUSE_MOVED:
+		case EMIE_LMOUSE_DOUBLE_CLICK:
+		case EMIE_RMOUSE_DOUBLE_CLICK:
+		case EMIE_MMOUSE_DOUBLE_CLICK:
+		case EMIE_LMOUSE_TRIPLE_CLICK:
+		case EMIE_RMOUSE_TRIPLE_CLICK:
+		case EMIE_MMOUSE_TRIPLE_CLICK:
+		case EMIE_COUNT:
```

Comment:
Would a `default:` not suffice?

Reply:
It should, I just didn't recompile after adding a `default`, which lead me to think that it didn't. Will fix it.